scripts/new_python_based_image.py

```diff
@@ -27,40 +29,32 @@ def configure_logger(log_level):
     LOGGER.setLevel(log_level)
 
 
+@dataclass
 class Args:
     """
     Class to encapsulate command-line arguments.
-
-    Attributes:
-        context_dir (str): The directory to use as the context for searching.
-        source (str): The source Python version.
-        target (str): The target Python version.
-        match (str): The string to match with the paths.
-        log_level (str): The logging level.
     """
-
-    def __init__(self, context_dir, source, target, match, log_level):
-        self.context_dir = context_dir
-        self.source = source
-        self.target = target
-        self.match = match
-        self.log_level = log_level
+    context_dir: str
+    source: str
+    target: str
+    match: str
+    log_level: str
 
     def __str__(self):
```

Comment:
https://docs.python.org/3/library/dataclasses.html#dataclasses.dataclass generates __repr__ method for you, is __str__ still necessary?

Reply:
The output from `__repr__` is not so pretty but I can live with that. Removed.